### PR TITLE
oauth: adds user_id and patron_info

### LINKED ISSUE #3710: Error when displaying a loan if the item has been deleted
### Bug description:
If I delete an item that was linked to a loan, the serialization creates an error if I try to see the loan in the API. 
Usually we don't use it but it still creates an error in the API.

Moreover, this can create problems in the interface when displaying past fees linked to this loan (error in the console).

### Expected behavior:
- The serialization for a loan should take into account that the item/document may not exist anymore and say "unknown" when it needs info that doesn't exist anymore OR implement soft delete

### Steps to Reproduce:
1. Find a loan with a fee
2. Checkin the item and "pay" the fee
3. Delete the item
4. Go to the patron profile and check fee history /professional/circulation/patron/[pid]/fees
5. See console and Sentry error

Example: https://bib.rero.ch/api/loans/412553

### LINKED ISSUE #3778: OAuth: return patron_pid in the scopes
### How it works

Certain providers need a pid to identify a user and link them to their data (loans, reservations, etc.). Our OAuth implementation returns a barcode but this field can change.

### Improvement suggestion

Allow the OAuth server to return the patron.pid somewhere in the scopes so that external services can specifically identify each user.

### PR DESCRIPTION
**Pull request by rerowep**
* Adds `user_id`.
* Adds `patron_info` scope with patron pid, patron type, institution
  and expiration_date.
* Marks `patron_types` as deprecated.
* Closes #3778.

### CODE REVIEW

File: rero_ils/modules/patrons/views.py [hunk 1]
```diff
@@ -371,6 +371,13 @@ def get_institution_code(institution):
 
     # Get the main patron
     patron = get_main_patron(patrons)
+    # Patrin pid
+    from pprint import pprint
+
+    print(">>>>>>>")
+    pprint(token_scopes)
```
Comment (PascalRepond) [inline rero_ils/modules/patrons/views.py:378]: Oops

**Review by PascalRepond — CHANGES_REQUESTED** [review]

**Review by PascalRepond — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -319,9 +319,11 @@ vendors = "rero_ils.modules.vendors.jsonschemas"
 files = "rero_ils.modules.files.jsonschemas"
 
 [tool.poetry.plugins."invenio_oauth2server.scopes"]
+fullname = "rero_ils.oauth.scopes:fullname"
 birthdate = "rero_ils.oauth.scopes:birthdate"
+patrons = "rero_ils.oauth.scopes:patrons"
```
Comment (PascalRepond) [inline pyproject.toml:324]: Maybe even call this scope "patron_info" to be even more clear

File: rero_ils/modules/patrons/views.py [hunk 3]
```diff
@@ -341,21 +331,10 @@ def get_institution_code(institution):
...
-    # Process for all patrons
-    patrons = copy.deepcopy(current_patrons)
-    for patron in patrons:
-        patron["institution"] = patron.organisation
-        patron["patron"]["type"] = PatronType.get_record_by_pid(
-            extracted_data_from_ref(patron["patron"]["type"]["$ref"])
-        )
+    # user = User.get_record(current_user.id).dumps_metadata()
 
     # Birthdate
```
Comment (PascalRepond) [inline rero_ils/modules/patrons/views.py:336]: Move the comment

**Comment by PascalRepond** [comment]
Please update your commit message to explain what we did. This doesn't close #3710.

File: rero_ils/modules/patrons/views.py [hunk 4]
```diff
@@ -364,35 +343,50 @@ def get_institution_code(institution):
...
-    if patron.get("patron", {}).get("barcode"):
-        data["barcode"] = patron["patron"]["barcode"][0]
+    patrons = current_patrons
+    if len(patrons) > 0:
+        patron = patrons[0]
+        # Barcode
+        if patron.get("patron", {}).get("barcode"):
+            data["barcode"] = patron["patron"]["barcode"][0]
     # Patron types
     if "patron_types" in token_scopes:
```
Comment (PascalRepond) [inline rero_ils/modules/patrons/views.py:357]: if "patron_info" in token_scopes ?

File: rero_ils/modules/patrons/views.py [hunk 5]
```diff
@@ -364,35 +343,50 @@ def get_institution_code(institution):
...
+            if expiration_date := patron.get("patron", {}).get("expiration_date"):
                 info["expiration_date"] = datetime.datetime.strptime(
-                    patron["patron"]["expiration_date"], "%Y-%m-%d"
+                    expiration_date, "%Y-%m-%d"
                 ).isoformat()
-            if info:
-                patron_types.append(info)
+            patron_types.append(info)
+            # new dict
+            if org := patron.organisation:
```
Comment (PascalRepond) [inline rero_ils/modules/patrons/views.py:376]: if "institution" in token_scopes ?

File: rero_ils/modules/patrons/views.py [hunk 6]
```diff
@@ -364,35 +343,50 @@ def get_institution_code(institution):
...
+                org_code = get_institution_code(org)
+                patron_info = {"patron_pid": patron.pid, "institution": org_code}
+                if patron_type_code := patron_type.get("code"):
+                    patron_info["patron_type"] = patron_type_code
+                if expiration_date := patron.get("patron", {}).get("expiration_date"):
```
Comment (PascalRepond) [inline rero_ils/modules/patrons/views.py:381]: if "expiration_date" in token_scopes ?

**Review by PascalRepond — CHANGES_REQUESTED** [review]

File: rero_ils/modules/patrons/views.py (hunk 5 above)
Comment (PascalRepond) [reply rero_ils/modules/patrons/views.py:376]: if "patron_info" in token_scopes ?

File: rero_ils/modules/patrons/views.py [hunk 7]
```diff
@@ -321,78 +320,83 @@ def info():
...
-            patron_type_code = patron.get("patron", {}).get("type", {}).get("code")
-            if patron_type_code:
+    patrons = current_patrons
+    if len(patrons) > 0:
+        patron = patrons[0]
+        # Barcode
+        if patron.get("patron", {}).get("barcode"):
+            data["barcode"] = patron["patron"]["barcode"][0]
+    # Patron
+    patron_pids = []
```
Comment (PascalRepond) [inline rero_ils/modules/patrons/views.py:359]: Is this variable really needed?

**Review by PascalRepond — APPROVED** [review]

File: rero_ils/modules/patrons/views.py (hunk 7 above)
Comment (rerowep) [reply rero_ils/modules/patrons/views.py:359]: Nop

**Review by Garfield-fr — APPROVED** [review]

File: rero_ils/modules/patrons/views.py [hunk 8]
```diff
@@ -321,78 +320,81 @@ def info():
...
-
         :param institution: Institution object.
         :returns: Code for the institution.
         """
-        # TODO: make this non rero specific using a configuration
-        return institution["code"] if institution["code"] != "nj" else "rbnj"
-
-    user = User.get_record(current_user.id).dumps_metadata()
+        code = institution["code"]
+        if changed_code := current_app.config.get(
```
Comment (jma) [inline rero_ils/modules/patrons/views.py:330]: Why an if?

File: rero_ils/modules/patrons/views.py [hunk 9]
```diff
@@ -321,78 +320,81 @@ def info():
...
-        data["barcode"] = patron["patron"]["barcode"][0]
-    # Patron types
-    if "patron_types" in token_scopes:
-        patron_types = []
-        for patron in patrons:
-            info = {}
-            patron_type_code = patron.get("patron", {}).get("type", {}).get("code")
-            if patron_type_code:
+    patrons = current_patrons
+    if len(patrons) > 0:
```
Comment (jma) [inline rero_ils/modules/patrons/views.py:353]: if patrons: should be enough

File: tests/api/patrons/test_patrons_rest.py [hunk 10]
```diff
@@ -545,31 +545,51 @@ def test_patron_info(app, client, patron_martigny, librarian_martigny):
...
+                "institution": "ORG1",
+                "patron_pid": patron_martigny.pid,
                 "patron_type": "patron-code",
             }
         ],
+        "patron_info": {
+            "ORG1": {
+                "expiration_date": patron_martigny["patron"]["expiration_date"]
+                + "T00:00:00",
+                "institution": "ORG1",
```
Comment (jma) [inline tests/api/patrons/test_patrons_rest.py:576]: institution name or institution pid?

**Review by jma — APPROVED** [review]

File: tests/api/patrons/test_patrons_rest.py (hunk 10 above)
Comment (rerowep) [reply tests/api/patrons/test_patrons_rest.py:576]: Institution is the pid transformable with `RERO_ILS_APP_INSTITUTION_CODE_TRANSFORMATION`

**Pull request merged**